### DISCUSSION TITLE
실제 코딩하면서 클로져를 많이 사용하나요? 주로 어떤 패턴으로 사용하나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#93] [클로저]
- 실제 코딩하면서 클로져를 많이 사용하나요? 주로 어떤 패턴으로 사용하나요?

**Comment by pipisebastian** [comment]
> 실제 코딩하면서 클로져를 많이 사용하나요? 주로 어떤 패턴으로 사용하나요?

- 실제로 클로져를 사용해본 경험은 없지만, 데이터 은닉이 필요할때 사용할 것 같습니다.
- 모듈 패턴을 사용하는 것으로 알고 있는데, 변수/함수를 즉시실행함수로 감싸, 하나의 모듈을 만드는 방식입니다. 외부에서 접근할 수 없고, 모듈에서 제공하는 메서드를 통해서만 접근할 수 있습니다.

**Comment by publdaze** [comment]
> - 실제 코딩하면서 클로져를 많이 사용하나요? 주로 어떤 패턴으로 사용하나요?

- 클로져 개념에 대해서 학습한 지 얼마되지 않아 실제 코딩할 때 많이 사용해본 경험은 없지만
- 정보은닉이 필요할 때, 유용하게 사용할 수 있을 것 같다.
- 정보은닉의 경우 es6부터 제공해주는 클래스를 활용할 수 있는 것으로 아는데,
- 클래스에서는 클로져 내부 메서드를 사용할 때 발생할 수 있는 인스턴스 메서드와 프로토타입 메서드의 한계를 해결해줄 수 있는 것으로 알아서
- 클래스에 비해 클로져를 사용하는 것의 장점이 있는 지에 대한 의문점이 있는데 이에 대해서는 추후 더 공부해보고 싶다.
- 클로져의 패턴으로는 모듈 패턴을 활용할 수 있을 것 같은데 즉시 실행 함수를 사용한 모듈 패턴을 통해 클로저를 많이 구현하는 걸로 알고있다.
